### PR TITLE
Dovzhyk Volodymyr HW 8

### PR DESCRIPTION
**Pull request by ReSpekTeR**

### CODE REVIEW

File: dovzhyk/HW8/hw8_task_1.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    elif operation == '/':
+        return num1/num2
+    else:
+        return 'Unknown operation'
+
+
+enterNum_1 = int(input('Enter first number:' ))
+enterNum_2 = int(input('Enter second number:' ))
+enterOperation = input('Enter sign of operation:' )
+print(simpleCalc(enterNum_1,enterNum_2,enterOperation))
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_1.py:17]: Good Way

File: dovzhyk/HW8/hw8_task_2.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def simpleDeposit(sum,period):
+    for i in range(period):
+        sum = sum + sum * 0.1
+    return sum
+
+
+depositAmount = int(input('Enter sum of deposit:' ))
+period = int(input('Enter years:' ))
+print(simpleDeposit(depositAmount,period))
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_2.py:9]: Nice and optimized

File: dovzhyk/HW8/hw8_task_3.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    if month == 2:
+        maxDay = 29  if year % 4 == 0 and year % 100 != 0 or year % 400 == 0 else 28
+    else:
+        maxDay = 30
+    return True if day in range(1, maxDay+1) else False
+
+day = int(input())
+month = int(input())
+year = int(input())
+print(checkData(day,month,year))
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_3.py:14]: Does this solution check the correctness of month?

File: dovzhyk/HW8/hw8_task_4.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+def is_prime(num):
+    result = True
+    if num > 5 and num % 10 == 5 or num > 2 and num % 2 == 0:
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_4.py:3]: Why did you separate this checking?

File: dovzhyk/HW8/hw8_task_4.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+while True:
+    enter = int(input('Enter the number more than 1: '))
+    if enter == 0:
+        break
+    if is_prime(enter) == True:
+        print(enter, 'is prime number')
+    else :
+        print(enter, 'is composite number.')
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_4.py:19]: Decision works. Nice

File: dovzhyk/HW8/hw8_task_4.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+def is_prime(num):
+    result = True
+    if num > 5 and num % 10 == 5 or num > 2 and num % 2 == 0:
+        result = False
+    else:
+        for i in range(2,num):
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_4.py:6]: The range could be more optimal, for example, range(2, num // 2 + 1)

File: dovzhyk/HW8/hw8_task_5.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+def is_year_leap(year):
+	return True if year % 4 == 0 and year % 100 != 0 or year % 400 == 0 else False
+
+
+while True:
+    enter = int(input())
+    if enter == 0:
+        break
+    else:
+        print(is_year_leap(enter))
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_5.py:10]: Nice and optimal decision

**Review by katemalash — CHANGES_REQUESTED** [review]
Please fix issue with month

File: dovzhyk/HW8/hw8_task_4.py (hunk 6 above)
Comment (ReSpekTeR) [reply dovzhyk/HW8/hw8_task_4.py:6]: Thanks for the tip.

File: dovzhyk/HW8/hw8_task_3.py (hunk 3 above)
Comment (ReSpekTeR) [reply dovzhyk/HW8/hw8_task_3.py:14]: No, I miss that moment

File: dovzhyk/HW8/hw8_task_4.py (hunk 4 above)
Comment (ReSpekTeR) [reply dovzhyk/HW8/hw8_task_4.py:3]: Hmm. Branching is not needed there at all.
I'll take a look at it!

File: dovzhyk/HW8/hw8_task_3.py [hunk 8]
```diff
@@ -1,14 +1,17 @@
...
-print(checkData(day,month,year))
\ No newline at end of file
+
+usersDay = int(input('Enter the day '))
+usersMonth = int(input('Enter the month '))
+usersYear = int(input('Enter the year '))
+if check_date(usersDay, usersMonth, usersYear):
+    print('The date is correct')
+else:
+    print('The date is incorrect')
```
Comment (katemalash) [inline dovzhyk/HW8/hw8_task_3.py:17]: Looks Good now.

**Review by katemalash — APPROVED** [review]
Good Job. You can merge pull-request

**Pull request merged**